app/server/lib/ProxyAgent.ts

```diff
@@ -18,7 +18,30 @@ export function proxyAgent(requestUrl: URL): HttpProxyAgent | HttpsProxyAgent |
     } catch (e) {
       // Include info helpful for diagnosing issues (but not the potentially sensitive full requestUrl).
       log.rawWarn(`ProxyAgent error ${e}`,
-        {proxy, reqProtocol: requestUrl.protocol, requestOrigin: requestUrl.origin});
+        {proxy: httpsProxy, reqProtocol: requestUrl.protocol, requestOrigin: requestUrl.origin});
+      throw e;
+    }
+  };
+  return agent;
+}
+
+export function proxyAgentForUntrustedRequests(requestUrl: URL): HttpProxyAgent | HttpsProxyAgent | undefined {
```

Comment:
My guess is that the code of `proxyAgentForTrustedRequests` and of `proxyAgentForUntrustedRequests` could be factorized. You may create a function internal to the module (which is not exported).

What do you think?